src/perf/lib/RpsClient.cpp

```diff
@@ -397,7 +399,10 @@ RpsConnectionContext::StreamCallback(
     case QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE:
         Worker->Client->StreamContextAllocator.Free(StrmContext);
         MsQuic->StreamClose(StreamHandle);
-        Worker->QueueSendRequest();
+        if (CxPlatTimeDiff32(Worker->Client->StartTime, CxPlatTimeMs32()) 
+            < (Worker->Client->RunTime + 3000)) {
```

Comment:
Still think `3000` is too much. No more than `1000` but I think `100` is also good.